### PR TITLE
Bugfix: Could not find an order by OrderId after it's OrderId was changed through the API

### PR DESCRIPTION
**Pull request by woutersamaey**
As discussed with @NicolasDorier

### CODE REVIEW

**Review by bolatovumar — APPROVED** [review]

File: BTCPayServer/Services/Invoices/InvoiceRepository.cs [hunk 1]
```diff
@@ -462,7 +470,24 @@ public async Task<InvoiceEntity> UpdateInvoiceMetadata(string invoiceId, string
...
+            var newMetadata = InvoiceMetadata.FromJObject(metadata);
+            var oldOrderId = blob.Metadata.OrderId;
+            var newOrderId = newMetadata.OrderId;
+
+            if (newOrderId != null)
+            {
+                // OrderId is saved in 2 places: (1) the invoice table and (2) in the metadata field. We are updating both for consistency.
+                invoiceData.OrderId = newOrderId;
+                
+                if (oldOrderId != null && !newOrderId.Equals(oldOrderId, StringComparison.InvariantCulture))
```
Comment (Kukks) [inline BTCPayServer/Services/Invoices/InvoiceRepository.cs:483]:
```suggestion
                if (oldOrderId != null && (newOrderId is null || !newOrderId.Equals(oldOrderId, StringComparison.InvariantCulture)))
```

File: BTCPayServer/Services/Invoices/InvoiceRepository.cs [hunk 2]
```diff
@@ -462,7 +470,24 @@ public async Task<InvoiceEntity> UpdateInvoiceMetadata(string invoiceId, string
...
+                {
+                    RemoveFromTextSearch(context, invoiceData, oldOrderId);
+                }
+                AddToTextSearch(context, invoiceData, new[] { newOrderId });
```
Comment (Kukks) [inline BTCPayServer/Services/Invoices/InvoiceRepository.cs:487]:
```suggestion
                if (newOrderId != null) {
                	AddToTextSearch(context, invoiceData, new[] { newOrderId });
                }
```

**Review by Kukks — CHANGES_REQUESTED** [review]
W

**Comment by woutersamaey** [comment]
Thanks for the improvements @Kukks

**Pull request merged**